src/v/archival/ntp_archiver_service.cc

```diff
@@ -2041,7 +2042,10 @@ ntp_archiver::maybe_truncate_manifest() {
             vlog(
               ctxlog.debug,
               "archival metadata STM update passed, re-uploading manifest");
-            co_await upload_manifest(sync_local_state_ctx_label);
+
+            // TODO(Lazin): Explain why it is safe to ignore manifest upload
+            // result.
+            std::ignore = co_await upload_manifest(sync_local_state_ctx_label);
```

Comment:
The manifest upload will be retried after some time because the STM will have 'dirty' flag set. I think that the `upload_manifest` will log an error so it won't be invisible.